### PR TITLE
Allocator against new isolation service

### PR DESCRIPTION
**Pull request by rostyboost**

### CODE REVIEW

File: titus_isolate/allocate/constants.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 TITUS_ISOLATE_CELL_HEADER = "X-Titus-Isolate-Cell"
 UNKNOWN_CELL = "unknown_cell"
 CELL = "cell"
+GRPC_CELL = "grpc_cell"
```
Comment (gabrielhartmann) [inline titus_isolate/allocate/constants.py:29]: Why is this needed?  Shouldn't a client land in a particular cell regardless of implementation?  I'd expect the `get_cell` function to determine which API it should address and return the correct cell.

File: titus_isolate/utils.py [hunk 2]
```diff
@@ -181,6 +185,26 @@ def get_cell_name():
         log.error("Failed to determine isolation cell.")
         return UNKNOWN_CELL
 
+def get_grpc_cell_name():
```
Comment (gabrielhartmann) [inline titus_isolate/utils.py:188]: Same comment as for the GRPC_CELL tag.

File: titus_isolate/allocate/remote/allocator.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+from titus_isolate.kub.constants import *
+from titus_isolate.kub.utils import get_node
+from titus_isolate.model.processor.config import get_cpu_from_env
+from titus_isolate.utils import get_config_manager
+
+
+ALLOCATOR_NAME = "GrpcRemoteIsolationAllocator"
+REQ_TYPE_METADATA_KEY = "req_type"
+
+class Allocator(CpuAllocator):
```
Comment (gabrielhartmann) [inline titus_isolate/allocate/remote/allocator.py:27]: Can we please name this something a bit less generic?
```
GrpcAllocator
```
maybe

File: titus_isolate/config/constants.py [hunk 4]
```diff
@@ -44,6 +44,11 @@
 REMOTE_FREE_ALLOCATOR = 'REMOTE_FREE_ALLOCATOR'
 REMOTE_REBALANCE_ALLOCATOR = 'REMOTE_REBALANCE_ALLOCATOR'
 
+# New Remote Allocator
+GRPC_REMOTE_ALLOC_ENDPOINT = 'TITUS_ISOLATE_GRPC_REMOTE_ALLOCATOR_ENDPOINT'
```
Comment (gabrielhartmann) [inline titus_isolate/config/constants.py:48]: You probably want all 3 of these in the "restart" list at the bottom of this file.

File: titus_isolate/utils.py [hunk 5]
```diff
@@ -181,6 +185,26 @@ def get_cell_name():
...
+    config_manager = get_config_manager()
+    if config_manager is None:
+        log.warning("Config manager is not yet set.")
+        return UNKNOWN_CELL
+
+    endpoint = config_manager.get_str(GRPC_REMOTE_ALLOC_ENDPOINT, None)
```
Comment (gabrielhartmann) [inline titus_isolate/utils.py:194]: If this function continues to exist, you probably want `get_cached_str` along with the inclusion in the "restart" list.

File: titus_isolate/allocate/remote/allocator.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        ctx.stack = node.metadata.annotations.get(ANNOTATION_KEY_STACK, '')
+        ctx.cluster = node.metadata.annotations.get(ANNOTATION_KEY_CLUSTER, '') 
+        ctx.autoscale_group = node.metadata.annotations.get(ANNOTATION_KEY_ASG, '')
+        ctx.resource_pool = node.metadata.annotations.get(LABEL_KEY_RESOURCE_POOL, '')
+        ctx.instance_type = node.metadata.annotations.get(ANNOTATION_KEY_INSTANCE_TYPE, '')
+        return ctx
+
+    def __init__(self, free_thread_provider):
+        config_manager = get_config_manager()
+        self.__endpoint = config_manager.get_str(GRPC_REMOTE_ALLOC_ENDPOINT, None)
```
Comment (gabrielhartmann) [inline titus_isolate/allocate/remote/allocator.py:48]: `get_cached_str`

File: titus_isolate/isolate/utils.py [hunk 7]
```diff
@@ -7,11 +7,12 @@
 from titus_isolate.allocate.noop_allocator import NoopCpuAllocator
 from titus_isolate.allocate.noop_reset_allocator import NoopResetCpuAllocator
 from titus_isolate.allocate.remote_cpu_allocator import RemoteCpuAllocator
+from titus_isolate.allocate.remote.allocator import Allocator as GrpcRemoteIsolServiceAllocator
```
Comment (gabrielhartmann) [inline titus_isolate/isolate/utils.py:10]: This is a 3rd name for the same allocator I think.

**Review by gabrielhartmann — CHANGES_REQUESTED** [review]
Please see comments.  I assume the core functionality of calling the GRPC service has been verified.  As long as we can return to using the current remote allocator, and dynamically configure the new GRPC remote allocator we should be good to go.

**Comment by rostyboost** [comment]
Thanks @gabrielhartmann . I believe I addressed all comments in https://github.com/Netflix-Skunkworks/titus-isolate/pull/337/commits/354dc913a6ede8595ecbde98b99ce0ed693d70d0

**Review by gabrielhartmann — APPROVED** [review]

**Pull request merged**